PopPUNK/utils.py

```diff
@@ -497,3 +494,162 @@ def isolateNameToLabel(names):
     # want to remove certain characters
     labels = [name.split('/')[-1].split('.')[0] for name in names]
     return labels
+
+
+def sketchlib_assembly_qc(assemblyList, prefix, klist, estimated_length,
+                 qc_filter, retain_failures, length_sigma, lower_length, upper_length, prop_n,
+                 upper_n, strand_preserved, threads):
+    """Calculates random match probability based on means of genomes
+    in assemblyList, and looks for length outliers.
+
+    Args:
+        assemblyList (str)
+            File with locations of assembly files to be sketched
+        prefix (str)
+            Prefix of output files
+        klist (list)
+            List of k-mer sizes to sketch
+        estimated_length (int)
+            Estimated length of genome, if not calculated from data
+        qc_filter (string)
+            Behaviour on identifying sequences failing QC
+        retain_failures (bool)
+            Keep sketches of genomes that fail QC separate from main
+            database
+        length_sigma (int)
+            Number of SDs of length distribution beyond which sequences
+            are excluded
+        lower_length (int)
+            Threshold length below which sequences are excluded
+        upper_length (int)
+            Threshold length above which sequences are excluded
+        prop_n (float)
+            Proportion of ambiguous bases above which sequences are excluded
+        upper_n (int)
+            Number of ambiguous bases above which sequences are excluded
+        strand_preserved (bool)
+            Ignore reverse complement k-mers (default = False)
+        threads (int)
+            Number of threads to use in parallelisation
+
+    Returns:
+        genome_length (int)
+            Average length of assemblies
+    """
+    # Genome length needed to calculate prob of random matches
+    if estimated_length is not None:
+        mean_genome_length = estimated_length # assume 2 Mb in the absence of other information
```

Comment:
This is a confusing way round to define this. Is this ever needed? Length should always be known/stored in the DB

Reply:
Removed `--estimated-length` in b1b0457.